frontend/components/elements/CourseItem.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+    <li class="course-item card">
+        <div class="card-body">
+            <template v-if="!editing">
+                <BButton
+                    size="sm"
+                    variant="outline-danger"
+                    class="float-right ml-1"
+                    v-confirm="{ action: deleteCourse, text: 'Are you sure you want to delete this course?' }">
+
+                    <VIcon name="trash"/>
+                </BButton>
+
+                <BButton
+                    size="sm"
+                    variant="outline-primary"
+                    class="float-right ml-1"
+                    @click="startEditing">
+
+                    <VIcon name="pencil"/>
+                </BButton>
+
+                <router-link
+                    :to="`/courses/${data.id}-${slugify(data.title)}`"
+                    class="h4 card-title link-unstyled">
+                    {{ data.short_title }}
+                </router-link>
+
+                <h5 class="h6 text-muted card-subtitle mt-0">
+                    {{ data.title }} ({{ getSemesterInfo(data.semester_type, data.semester_year).fullString }})
+                </h5>
+
+                <p
+                    class="card-text mt-2"
+                    v-if="data.hyperlink">
+
+                    <a :href="data.hyperlink">{{ data.hyperlink }}</a>
+                </p>
+
+                <router-link
+                    :to="`/courses/${data.id}-${slugify(data.title)}`"
+                    class="btn btn-primary">
+                    View projects, topics and sprints
+                </router-link>
+            </template>
+
+            <CourseForm
+                v-else
+                no-semester
+                v-model="editingData"
+                @cancel="editing = false"
+                @submit="saveCourse"/>
```

Comment:
I cannot save a course, after editing. Does not matter whether I am editing a field or not. There are no error messages.

Reply:
@anjakammer ah shoot, I forgot to adjust the semester validation again after fixing that. should work now.